File: eyetronics project Sam/UNITY/unity-facebook test/eyeColorMatching/Web_Files/api/phphandler.php
<?php
//============================================================================
// Title:  Color Matching (Unity-Facebook)
//----------------------------------------------------------------------------
// File:   phphandler.php
// Author: Nick Breslin (nickbreslin@gmail.com), Waddlefarm.org
//
// Copyright (c) Nick Breslin, 2009-2010. All Rights Reserved.
//----------------------------------------------------------------------------
//
// * This license notice may not be removed or altered.
// * All project files are free for non-commercial or commercial use.
// * I appreciate personal notification if any part of this project is used.
// * All profile files are provided "as is". No support or compatibility is to
//    assumed. I assume no responsible for results from using these files.
// * Donations are appreciated, please visit: http://www.waddlefarm.org/donate
//
//============================================================================

  
require( "database.php" );


$action    = $_POST[ 'action' ];
$unityHash = $_POST[ 'hash'   ];

$phpHash = md5( "savabhashkey" );

if( $phpHash != $unityHash )
	exit;


switch ( $action )
{
	case "EN_Ping":
		Ping();
		break;
		
	case "EN_GetUser":
		GetUser( $link );
		break;

	case "EN_PutScore":
		PutScore( $link );
		break;
		
	case "EN_GetStats":
		GetStats( $link );
		break;
		
	case "EN_GetFriends":
		GetFriends( $link );
		break;
		
	case "EN_Log":
		LogError( $link );
		break;
	
		
	default:
		return;
}



//============================================================================
// function Ping()
//----------------------------------------------------------------------------
// Tests connection.
//============================================================================
function Ping()
{
	echo "Test.";
}


//============================================================================
// function GetUser()
//----------------------------------------------------------------------------
// Gets User Data by ID.
//============================================================================
function GetUser( $link )
{		
	$fb_id = $_POST[ "id" ];
		
	$query = "SELECT * FROM fb_users WHERE fb_id = '$fb_id'";		
	$results = mysql_query( $query, $link ) or die( mysql_error() );

	if( mysql_num_rows( $results ) > 0 ) 
	{	
		while( $result = mysql_fetch_array( $results ) )
		{
			$record  =       $result[ 'fb_id'   ];	
			$record .= ';' . $result[ 'f_name'  ];
			$record .= ';' . $result[ 'pic_url' ];
			$record .= ';' . $result[ 'score'   ];
		}
		
		echo $record;
	}
	else
		echo "N/A";

}


//============================================================================
// function PutScore()
//----------------------------------------------------------------------------
// Updates score value for player.
//============================================================================
function PutScore( $link )
{			 
	$fb_id = $_POST[ 'id'    ];
	$score = $_POST[ 'score' ];
		
	$query = "UPDATE fb_users SET score = '$score' WHERE fb_id = '$fb_id'";		

	mysql_query( $query, $link ) or die( mysql_error() );
	
	echo $fb_id;
}


//============================================================================
// function GetFriends()
//----------------------------------------------------------------------------
// Gets the Facebook IDs for friends of the user who have a score.
//============================================================================
function GetFriends( $link )
{
	$fb_id = $_POST[ 'id'    ];
	$query   = "SELECT fb_users.fb_id as fb_id FROM fb_users, fb_friends WHERE fb_users.score > 0 and fb_friends.fb_id = '$fb_id' AND fb_friends.friend_id = fb_users.fb_id ORDER BY fb_users.score";
	
	$results = mysql_query( $query, $link ) or die( mysql_error() );
	$friends = mysql_num_rows( $results );

	if( $friends > 0 ) 
	{	
		while( $result = mysql_fetch_array( $results ) )
		{	
			 $ids .= ";" . $result[ 'fb_id' ];
		}
	}
	//------------------------------------------------------------------------
	// The "X" is neccessary, and removed in the Unity app.
	//------------------------------------------------------------------------
	echo "X".$ids;
}


//============================================================================
// function GetStats()
//----------------------------------------------------------------------------
// Calculates database population and rank.
//============================================================================
function GetStats( $link )
{		
	$fb_id = $_POST[ "id" ];

	$query   = "SELECT fb_id, score FROM fb_users WHERE score > 0 ORDER BY score";		
	$results = mysql_query( $query, $link ) or die( mysql_error() );

	//------------------------------------------------------------------------
	// Count is the total number of user accounts with a score.
	//------------------------------------------------------------------------
	$count   = mysql_num_rows( $results );
	
	$globalRank = 1;
	$score = 0;
	
	//------------------------------------------------------------------------
	// GlobalRank starts at 1, and increases for each row until the user's row.
	//------------------------------------------------------------------------
	if( $count > 0 ) 
	{	
		while( $result = mysql_fetch_array( $results ) )
		{
			if( $fb_id == $result[ 'fb_id' ] )
			{
				$score = $result[ 'score' ];
				break;
			}
			
			$globalRank++;
		}

		$query   = "SELECT fb_users.fb_id as fb_id, fb_users.score as score FROM fb_users, fb_friends WHERE fb_users.score > 0 and fb_friends.fb_id = '$fb_id' AND fb_friends.friend_id = fb_users.fb_id ORDER BY fb_users.score";
		$results = mysql_query( $query, $link ) or die( mysql_error() );
		
		//------------------------------------------------------------------------
		// Friends is the total number of user accounts with a score who are
		// friends with the current user.
		//------------------------------------------------------------------------
		$friends = mysql_num_rows( $results );
	
		$localRank = 1;
	
		//------------------------------------------------------------------------
		// LocalRank starts at 1, and increases for each row until the user's row.
		//------------------------------------------------------------------------
		if( $friends > 0 && $score != 0 ) 
		{	
			while( $result = mysql_fetch_array( $results ) )
			{	
				if( $score > $result[ 'score' ] )
				$localRank++;
				
				if( $fb_id == $result[ 'fb_id' ] )
					break;
			}
		}
			
		$record .=       $count;
		$record .= ';' . $friends;
		$record .= ';' . $globalRank;
		$record .= ';' . $localRank;
		
		echo $record;
	}
}

mysql_close();
?>

File: eyetronics project Sam/UNITY/eyeBdancing/eyeBdancing/Assets/Facebook/Scripts/PHPHandler.cs
//============================================================================
// Title:  Color Matching (Unity-Facebook)
//----------------------------------------------------------------------------
// File:   PHPHandler.cs
// Author: Nick Breslin (nickbreslin@gmail.com), Waddlefarm.org
//
// Copyright (c) Nick Breslin, 2009. All Rights Reserved.
//----------------------------------------------------------------------------
//
// * This license notice may not be removed or altered.
// * All project files are free for non-commercial or commercial use.
// * I appreciate personal notification if any part of this project is used.
// * All project files are provided "as is". I assume no responsibility for
//   results from using these files, nor should any support be expected.
// * Donations are appreciated, please visit: http://www.waddlefarm.org/donate
//
//============================================================================

using UnityEngine;
using System.Collections;

//============================================================================
// class PHPHandler()
//----------------------------------------------------------------------------
// Manages interfacing between Unity and PHPHandler.php
//============================================================================
public class PHPHandler : MonoBehaviour
{

	
	//------------------------------------------------------------------------
	// enum Action
	//------------------------------------------------------------------------
	public enum Action
	{
		EN_Ping,
		EN_GetUser,
		EN_PutScore,
		EN_GetStats,
		EN_GetTxrUrl,
		EN_GetMeshUrl
	}


	//------------------------------------------------------------------------
	// enum Query
	//------------------------------------------------------------------------
	public enum Query
	{
		EN_Sending,
		EN_Success,
		EN_Fail
	}


	//------------------------------------------------------------------------
	// Globals, public
	//------------------------------------------------------------------------	
	public Director kDirector;
	public delegate void Callback( string data );
	public delegate void CallbackTexture( WWW data );
	public string sBaseUrl;
	public string sHash;
	public string sTestString;
	
	
	//------------------------------------------------------------------------
	// Globals, GUI Windows
	//------------------------------------------------------------------------
	[HideInInspector] public Query eQuery;
	
	
	//========================================================================
	// void Start()
	//------------------------------------------------------------------------
	// Performs empty check and begins connection test.
	//========================================================================
	void Awake()
	{	
		if( kDirector   == null ) Debug.LogError( "BaseUrl cannot be empty." );
		if( sBaseUrl    == ""   ) Debug.LogError( "BaseUrl cannot be empty." );
		if( sHash       == ""   ) Debug.LogError( "Hash cannot be empty."    );
		if( sTestString == ""   ) Debug.LogError( "Hash cannot be empty."    );

		//Encrypt Hash
		sHash = Md5Sum( sHash );
	}	


	//========================================================================
	// Start
	//------------------------------------------------------------------------
	// Pings the PHPHandler.php script/server.
	//========================================================================	
	public void Init()
	{
		StartCoroutine( Ping( true ) );
	}


	//========================================================================
	// void Request() -- Overloads
	//------------------------------------------------------------------------
	// These are the overload methods for our primary Request() method. They
	// allow for various ways to call the request, each passing the initial
	// call to another method, until finally the primary method.
	//========================================================================
	public void Request( Action eAction )
	{
		Request( eAction, null );
	}

	public void Request( WWWForm wForm, Action eAction )
	{
		Request( wForm, eAction, null );
	}
	
	public void Request( Action eAction, Callback callback )
	{
		WWWForm wForm = new WWWForm();
		Request( wForm, eAction, callback );
	}
	

	//========================================================================
	// void Request()
	//------------------------------------------------------------------------
	// Primary Request(). Attaches the action and the hash (security) into the
	// WWWForm and calls the coroutine, Handle().
	//========================================================================	
	public void Request( WWWForm wForm, Action eAction, Callback callback )
	{
		wForm.AddField( "action", "" + eAction ); 
		wForm.AddField( "hash", sHash );	

		StartCoroutine( Handle( new WWW( sBaseUrl, wForm ), callback ) );
	}


	//========================================================================
	// void RequestObj()
	//------------------------------------------------------------------------
	// Attaches the action and the hash (security) into the WWWForm and calls
	// the coroutine, HandleTexture().
	//========================================================================	
	public void RequestObj( string url, CallbackTexture callback )
	{	
		StartCoroutine( HandleObj( new WWW( url ), callback ) );
	}


	//========================================================================
	// void RequestTexture()
	//------------------------------------------------------------------------
	// Attaches the action and the hash (security) into the WWWForm and calls
	// the coroutine, HandleTexture().
	//========================================================================	
	public void RequestTexture( string url, CallbackTexture callback )
	{	
		StartCoroutine( HandleTexture( new WWW( url ), callback ) );
	}	


	//========================================================================
	// IEnumerator Handle()
	//------------------------------------------------------------------------
	// Handle is called as a coroutine, as the length of time to receive a
	// response from the PHP script is undetermined. A coroutine is needed so
	// the application does not hang. Executes a callback once the retrieval
	// of data is complete.
	//
	// Pass the data variable of the www to the callback.
	//========================================================================	
	public IEnumerator Handle( WWW www, Callback callback )
	{	
		yield return www;

    	if( callback != null )    
			callback( www.data );
		
		www.Dispose();
	}


	//========================================================================
	// IEnumerator HandleObj()
	//------------------------------------------------------------------------
	// Handle is called as a coroutine, as the length of time to receive a
	// response from the PHP script is undetermined. A coroutine is needed so
	// the application does not hang. Executes a callback once the retrieval
	// of data is complete.
	//
	// Pass the entire www result to the callback.
	//========================================================================	
	public IEnumerator HandleObj( WWW www, CallbackTexture callback )
	{	
		yield return www;

    	if( callback != null )  
			callback( www );
		
		www.Dispose();
	}


	//========================================================================
	// IEnumerator HandleTexture()
	//------------------------------------------------------------------------
	// Handle is called as a coroutine, as the length of time to receive a
	// response from the PHP script is undetermined. A coroutine is needed so
	// the application does not hang. Executes a callback once the retrieval
	// of data is complete.
	//
	// Pass the entire www result to the callback.
	//========================================================================	
	public IEnumerator HandleTexture( WWW www, CallbackTexture callback )
	{	
		yield return www;

    	if( callback != null )  
			callback( www );
		
		www.Dispose();
	}
	

	//========================================================================
	// IEnumerator Ping()
	//------------------------------------------------------------------------
	// Initiates Ping attempt and displays connection result to the Log.
	//========================================================================	
	IEnumerator Ping( bool bIsStart )
	{
		Log.Add( "Pinging Server..." );

		eQuery = PHPHandler.Query.EN_Sending;		
		Request( PHPHandler.Action.EN_Ping, PingResult );
		
		while( eQuery == PHPHandler.Query.EN_Sending )
			yield return new WaitForSeconds( 0 );			

		
		string sConnection = "Connection ";
		Log.Type eType = Log.Type.EN_Message;
		
		if( eQuery == PHPHandler.Query.EN_Fail )
		{
			sConnection += "Failed.";
			eType = Log.Type.EN_Error;
		}
		
		else if( eQuery == PHPHandler.Query.EN_Success )
			sConnection += "Established.";
		
		Log.Add( sConnection, eType );
		
		yield return new WaitForSeconds( 0 );
		
		if( bIsStart ) {
			kDirector.kUserManager.GetUserID();
		}
	}


	//========================================================================
	// void PingResult()
	//------------------------------------------------------------------------
	// Callback Method. Compares the retrieved data to a predetermined string.
	// This is done www.error only catches an connection problems, not 404 
	// page errors.
	//========================================================================	
	public void PingResult( string data )
	{
		if( data == sTestString )
			eQuery = Query.EN_Success;
			
		else
			eQuery = Query.EN_Fail;
	}
	

	//========================================================================
	// string Md5Sum()
	//------------------------------------------------------------------------
	// Converts a string to md5 encryption.
	//========================================================================
	public string Md5Sum( string strToEncrypt )
	{
	    System.Text.UTF8Encoding ue = new System.Text.UTF8Encoding();
	    byte[] bytes = ue.GetBytes(strToEncrypt);

	    System.Security.Cryptography.MD5CryptoServiceProvider md5 = new System.Security.Cryptography.MD5CryptoServiceProvider();
	    byte[] hashBytes = md5.ComputeHash( bytes );

	    string hashString = "";

	    for ( int i = 0; i < hashBytes.Length; i++ )
	        hashString += System.Convert.ToString( hashBytes[i], 16 ).PadLeft( 2, '0' );

	    return hashString.PadLeft( 32, '0' );
	}
}